### PR TITLE
[CI] Fix the integrations tests Dockerhub rate limit

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR fixes the integration tests Dockerhub rate limit.

### Motivation

The [agent-integrations-test](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/693638411) job is often failing with:
```
docker.errors.BuildError: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

### CODE REVIEW

**Review by chouetz — APPROVED** [review]
Should we backport this change?

**Comment by amenasria** [comment]
> Should we backport this change?
We can yes !

File: tasks/docker_tasks.py [hunk 1]
```diff
@@ -53,9 +53,9 @@ def dockerize_test(ctx, binary, skip_cleanup=False):
 
     with open(f"{temp_folder}/Dockerfile", 'w') as stream:
         stream.write(
-            """FROM ubuntu:20.04
-COPY --from=docker/compose-bin:v2.26.1 /docker-compose /usr/bin/compose
-COPY --from=docker:26.1-cli /usr/local/bin/docker /usr/bin/docker
+            """FROM public.ecr.aws/docker/library/ubuntu:20.04
+COPY --from=public.ecr.aws/docker/library/docker:26.1-cli /usr/local/bin/docker /usr/bin/docker
+RUN wget https://github.com/docker/compose/releases/download/v2.26.1/docker-compose-linux-x86_64 -O /usr/bin/compose
```
Comment (KevinFairise2) [inline tasks/docker_tasks.py:58]: Can we check the sha of the downloaded binary?
Something similar to that one: https://github.com/DataDog/datadog-agent/blob/main/.gitlab/deps_build/deps_build.yml#L15-L19

Comment (amenasria) [reply tasks/docker_tasks.py]: Good point, fixed in [b075a9e](https://github.com/DataDog/datadog-agent/pull/30699/commits/b075a9e05bf0d6a60474c6ea054cf396af0d3b4f)

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**